code/modules/telesci/telesci_computer.dm

```diff
@@ -390,8 +415,42 @@
 		eject()
 		temp_msg = "NOTICE:<BR>Bluespace crystals ejected."
 
+	if(href_list["toggle_device_mode"])
+		device_mode = !device_mode
+
+	if(href_list["edit_device"])
+		if(attached_device)
+			attached_device.attack_self(usr)
+	if(href_list["remove_device"])
+		attached_device.loc = get_turf(src)
+		attached_device:holder = null
```

Comment:
Pretty sure this line will work with just a `.` operator instead of `:` operator, but I'll point out the issue anyways;
The `:` operator is only acceptable in large loops where efficiency is essential: Otherwise, it bypasses all type saftey, which is a bad thing, and will fail silently instead of generating a type error, which is not good.